### ISSUE TITLE
CT / RGB / XY translation

### ISSUE THREAD

**Issue by atxbyea** [claytonjn/hass-circadian_lighting#45]
While I was still using the IKEA gateway I would adjust all lights (both CT and RGB) bulbs with just lights_ct and they were in sync.

However after testing out zigbee2mqtt and Hue controlling the IKEA bulbs the RGB bulbs refuse CT setting and need to be added to either light_rgb or light_xy , however the lights are way out of colour from eachother.

Would it be possible to make circadian lighting translate the CT to a RGB or XY value that makes the lights actually similar? 

I can supply examples and do testing if necessary.

**Comment by claytonjn** [comment]
Circadian Lighting uses the built-in Home Assistant utilities to convert CT to RGB and RGB to XY. 

The biggest problem with those conversions is that color temperature by definition is a measurement of light, but RGB and XY are measurements of color. The conversion between them are estimates to begin with. 

On top of that, the way each light interprets color temperature values vs RGB/XY values can be drastically different - this is especially true with RGBW bulbs. A good example would be CT 6600k, which converts to 255, 255, 255 (pure white) - with some lights sending the CT would cause the bulb to only light up the W diode but sending the RGB value would turn on the R, G, and B pixels (combining to create white). 

Basically, at the end of the day, anything I do from the CL side could improve things for one type of light and worsen things for another. The best solution would be to figure out why Zigbee2MQTT/Hue doesn't allow sending CT values for those bulbs.

**Comment by atxbyea** [comment]
Wonderful answer Clayton.

I will buy an additional RGB bulb from Tradfri and play around with the different hubs and update the issue if I figure out something.

**Comment by bryshaw** [comment]
Anything?

**Comment by claytonjn** [comment]
@bryshaw what are you asking about?

**Comment by bryshaw** [comment]
@claytonjn Wondering @atxbyea findings on the new bulbs.
